### DISCUSSION TITLE
Issue when trying to implement Multiple Language plugin

### DISCUSSION THREAD

**Discussion by SergioAlias** [greenelab/lab-website-template#281] [Q&A] [answered]
### Checks

- [x] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://github.com/mycroorg/mycroorg.github.io

### Version of Lab Website Template you are using

1.2.2

### Description

Hello! Thank you so much for the template. I'm trying to implement the [jekyll-multiple-languages-plugin](https://github.com/kurtsson/jekyll-multiple-languages-plugin) in order to display my website either in English or Spanish. I'm following instructions in the plugin GitHub README, and the installation step is essentially the instructions you provide in the [documentation](https://greene-lab.gitbook.io/lab-website-template-docs/advanced/jekyll-plugins) (i.e, add the plugin to the config file and the Gemfile). However, when I do this and commit to main, the GitHub action (on push to main) fails. The plugin README says I need to install the gem, and I suppose I should specify that in a new or existing GitHub Action, but it is my first time using Actions and I'm still not confident enough working with them.

I understand that implementation of non pre-installed plugins is potentially out of the scope of this Discussions facility,  and I don't pretend template maintainers to tell me exactly how can I do an optional thing on my website. I want to ask if I'm on the right track thinking that I need to change something in Actions, or if I would need to run the "first time setup" action again (that wouldn't be a big problem since the website does not have content at all for now), but I would like to know before I try because that would imply deleting the repo and generate it again.

Thank you so much in advance

**Comment by vincerubinetti** [comment, marked as answer]
There's something missing from the docs which is that you need to generate a new `Gemfile.lock` file for this change as well.

<details><summary>logs from your failed workflow</summary>

```
Run ruby/setup-ruby@v1.172.0
Modifying PATH
Downloading Ruby
Extracting  Ruby
Print Ruby version
Installing Bundler
> bundle install
/opt/hostedtoolcache/Ruby/3.1.4/x64/bin/bundle config --local path /home/runner/work/mycroorg.github.io/mycroorg.github.io/vendor/bundle
/opt/hostedtoolcache/Ruby/3.1.4/x64/bin/bundle config --local deployment true
Cache key: setup-ruby-bundler-cache-v6-ubuntu-22.04-x64-ruby-3.1.4-wd-/home/runner/work/mycroorg.github.io/mycroorg.github.io-with--without--only--Gemfile.lock-c6cc3cbb2889edf[7](https://github.com/mycroorg/mycroorg.github.io/actions/runs/11035029395/job/30650080055#step:4:8)cd87179dd56d156e98d76fafad9826cb8dd4b2c5432311a3
Cache Size: ~40 MB (41457990 B)
/usr/bin/tar -xf /home/runner/work/_temp/aef5a158-5a58-4358-bb61-7461d5b9ae4d/cache.tzst -P -C /home/runner/work/mycroorg.github.io/mycroorg.github.io --use-compress-program unzstd
Cache restored successfully
Found cache for key: setup-ruby-bundler-cache-v6-ubuntu-22.04-x64-ruby-3.1.4-wd-/home/runner/work/mycroorg.github.io/mycroorg.github.io-with--without--only--Gemfile.lock-c6cc3cbb2889edf7cd87179dd56d156e98d76fafad9826cb8dd4b2c5432311a3
/opt/hostedtoolcache/Ruby/3.1.4/x64/bin/bundle install --jobs 4
The dependencies in your gemfile changed, but the lockfile can't be updated
because frozen mode is set

You have added to the Gemfile:
* jekyll-multiple-languages-plugin

Run `bundle install` elsewhere and add the updated Gemfile to version control.
If this is a development machine, remove the Gemfile.lock freeze by running
`bundle config set frozen false`.
Error: The process '/opt/hostedtoolcache/Ruby/3.1.4/x64/bin/bundle' failed with exit code [16](https://github.com/mycroorg/mycroorg.github.io/actions/runs/11035029395/job/30650080055#step:4:20)
```

</details>

To do this, you should just need to [preview your site locally](https://greene-lab.gitbook.io/lab-website-template-docs/getting-started/preview-your-site#on-your-computer-locally), which should generate a new `Gemfile.lock` file that you can commit, and then the GitHub Actions should succeed.

So, either run `./.docker/run.sh` with Docker installed, or run `bundle install` if you have Ruby and the bundler gem installed.

**Reply by vincerubinetti** [reply]
I tested this out locally, and it added the following to my Gemfile.lock:

```
  specs:
    ...
    jekyll-multiple-languages-plugin (1.8.0)
      jekyll (>= 2.0, < 5.0)

DEPENDENCIES
  ...
  jekyll-multiple-languages-plugin
```

From there, the site was able to build/run, albeit with some errors from the plugin because I hadn't set up languages and other things it wanted me to set up.

**Reply by SergioAlias** [reply]
After generating the `Gemfile.lock` and meeting the plugin requirements (add languages to the config, create folder with a YAML for each language) the Action was successful! Thank you so much for the quick reply!
